INTMDB-856: Add cloudProviderConfig to mongodbatlas_federated_database_instance (#1215)
mongodb/terraform-provider-mongodbatlas: website/docs/r/federated_database_instance.html.markdown

-> **NOTE:** Groups and projects are synonymous terms. You may find group_id in the official documentation.

## Example Usages with MongoDB Atlas Cluster as storage database

```terraform
resource "mongodbatlas_federated_database_instance" "test" {
  project_id         = "PROJECT ID"
  name = "NAME OF THE FEDERATED DATABASE INSTANCE"
  storage_databases {
    name = "VirtualDatabase0"
    collections {
      name = "NAME OF THE COLLECTION"
      data_sources {
          collection = "COLLECTION IN THE CLUSTER"
          database = "DB IN THE CLUSTER"
          store_name =  "CLUSTER NAME"
      }
    }
  }

  storage_stores {
    name = "STORE 1 NAME"
    cluster_name = "CLUSTER NAME"
    project_id = "PROJECT ID"
    provider = "atlas"
    read_preference {
      mode = "secondary"
    }
  }
}
```

> ## Example Usages with AWS S3 bucjet as storage database

Reviewer: ```suggestion
## Example Usages with Amazon S3 bucket as storage database
```

Reviewer: small typo, also name of service is Amazon S3 (not AWS S3) - https://aws.amazon.com/s3/

  }
}
```
## Argument Reference

* `project_id` - (Required) The unique ID for the project to create a Federated Database Instance.
* `name` - (Required) Name of the Atlas Federated Database Instance.

>   ### `cloud_provider_config` - Cloud provider linked to this data federated instance.

Reviewer: ```suggestion
  ### `cloud_provider_config` - (Optional) Cloud provider linked to this data federated instance.
```

>   #### `aws` - (Required) AWS provider of the cloud service where the Federated Database Instance can access the S3 Bucket.

Reviewer: ```suggestion
  #### `aws` - (Required) AWS provider of the cloud service where the Federated Database Instance can access the S3 Bucket. Note this parameter is only required if using `cloud_provider_config` since AWS is currently the only supported Cloud vendor on this feature at this time. 
```

>   * `role_id` - (Required) Unique identifier of the role that the Federated Instance can use to access the data stores. If necessary, use the Atlas [UI](https://docs.atlas.mongodb.com/security/manage-iam-roles/) or [API](https://docs.atlas.mongodb.com/reference/api/cloud-provider-access-get-roles/) to retrieve the role ID. You must also specify the `aws.0.test_s3_bucket`.

Reviewer: is `aws.0.test_s3_bucket` a typo? i thought we flattened to remove the # (i.e. 0.)

Author: updated

  * `test_s3_bucket` - (Required) Name of the S3 data bucket that the provided role ID is authorized to access. You must also specify the `aws.0.role_id`.

>   ### `data_process_region` - (Optional) The cloud provider region to which the Federated Instance routes client connections for data processing.

Reviewer: looks like we are missing "SINGAPORE_SGP" and "SAOPAULO_BRA" regions, but instead of updating does it make sense to simple delete all and link docs to https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Data-Federation/operation/createFederatedDatabase to we don't have to keep updating when new regions are added in future?

Author: it makes sense. Updated
